### PR TITLE
add privacy policy

### PR DESCRIPTION
**Pull request by ronie**
as reported by @MartijnKaijser on trac (http://trac.kodi.tv/ticket/17016) it's mandatory to provide a way for users to read our privacy policy within kodi.

this pr adds this functionality to the systeminfo window.

please check if i got the install path on each platform correct, the file should be available in special://xbmc/
(assuming we want to save it there...)

privacypolicy

### CODE REVIEW

**Comment by ronie** [comment]
todo: read the file only once, on window open (instead of each time the button is focused)

File: privacy-policy.txt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+We learn information about you when:
+
+    • you give it to us directly (e.g., when you choose to send us logs in the forums);
+    • we collect it automatically through our software and services (e.g., when your Kodi connects with our servers to update add-ons, or in download statistics provided by Google Play);
+    • when we try and understand more about you based on information you’ve given to us (e.g., when we use your platform information provided to the add-on server to figure out which platforms use Kodi the most, to better understand where we should focus our efforts).
+
+
+[B]What do we do with your information once we have it?[/B]
+
+Generally, we use your information to help us provide and improve our software and services for you (e.g., we use a log you send us to figure out why Kodi isn’t playing a video right or why it might have crashed, or we determine how many active users are using each platform in order to determine how to allocate resources per platform).
```
Comment (da-anda) [inline privacy-policy.txt]: shouldn't it be `sent` instead of `send` in "e.g., we use a log you sent us to figure..."

Comment (ronie) [inline privacy-policy.txt]: i didn't proof-read it ;-)
it's a copy of our policy taken from here: https://kodi.tv/kodi-privacy-policy/

Comment (hudokkow) [inline privacy-policy.txt]: ping @jjd-uk to proof-read it

File: project/cmake/scripts/linux/Install.cmake [hunk 2]
```diff
@@ -132,6 +132,10 @@ install(FILES ${CORE_SOURCE_DIR}/copying.txt
         DESTINATION ${datarootdir}/doc/${APP_NAME_LC}
         COMPONENT kodi)
 
+install(FILES ${CORE_SOURCE_DIR}/privacy-policy.txt
+        DESTINATION ${datarootdir}${APP_NAME_LC}
+        COMPONENT kodi)
+
```
Comment (hudokkow) [inline project/cmake/scripts/linux/Install.cmake]: Any reason you didn't used the code block above and install it to ${datarootdir}/doc/${APP_NAME_LC}, same as the other files?

Comment (ronie) [inline project/cmake/scripts/linux/Install.cmake]: special://xbmc/ points to ${datarootdir}${APP_NAME_LC}
(i need a special path that's the same across platforms)

File: privacy-policy.txt (hunk 1 above)
Comment (garbear) [inline privacy-policy.txt]: The document uses simple present tense, so send is correct

File: privacy-policy.txt [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+
+[B]When do we share your information with others?[/B]
+
+    • When we have asked and received your permission to share it.
+    • When we are fulfilling our educational purpose. We sometimes publicly release information to make our software better and foster an open web, but when we do so, we will remove your personal information and try to disclose it in a way that minimizes the risk of you being re-identified. For example, here is a blog post where we attempt to analyze from available data how many active users of XBMC existed at the time and share that info with the community.
```
Comment (garbear) [inline privacy-policy.txt]: I suggest we change:

`For example, here is a blog post where we attempt to analyze...`

to:

`For example, in a blog post we might attempt to analyze...`

Also, XBMC -> Kodi

**Comment by garbear** [comment]
Overall, I think we have a good privacy policy. Good job guys.

File: privacy-policy.txt (hunk 3 above)
Comment (natethomas) [inline privacy-policy.txt]: Good change. The original language is based on having a link to a blog post, which we obviously can't have here.

Comment (ronie) [inline privacy-policy.txt]: both fixed now

**Comment by Hedda** [comment]
Should this privacy policy be translated via Transifex when shown in the GUI?

http://www.transifex.com/teamxbmc/kodi-main/

If not using English inside Kodi then could expect this to be translated too?

**Comment by MartijnKaijser** [comment]
@Hedda sorry but that would be a really bad idea cause as we noticed in general the translations are quite horribly done. As this is a "legal"  document you cannot trust regular users to translate it.

File: xbmc/windows/GUIWindowSystemInfo.cpp [hunk 4]
```diff
@@ -199,3 +219,16 @@ void CGUIWindowSystemInfo::SetControlLabel(int id, const char *format, int label
...
+{
+  CFile file;
+  XFILE::auto_buffer buf;
+  if (file.LoadFile("special://xbmc/privacy-policy.txt", buf) > 0)
+  {
+    std::string strBuf(buf.get(), buf.length());
+    return strBuf;
+  }
+  else
+    return "";
```
Comment (tamland) [inline xbmc/windows/GUIWindowSystemInfo.cpp]: Please *always* log errors;)

**Comment by tamland** [comment]
I think this should rather be added as an info label (similar to all the other system info). Loading a file in CGUIWindowSystemInfo is going to cause it to lag, and the privacy policy will be coupled to and only viewable in this specific window.

**Comment by fetzerch** [comment]
For android this has to be added to https://github.com/xbmc/xbmc/blob/master/tools/android/packaging/Makefile.in#L111 i think. But I'm not sure which directory it needs to go inside the apk. maybe @koying can help here?

**Comment by ronie** [comment]
@tamland moved the functionality to CSysInfo and made it available as an infolabel.
please check if this is what you've had in mind.

**Comment by MartijnKaijser** [comment]
Now also works on Android. 
Any chance of making it possible to scroll through the text manually?

**Comment by MartijnKaijser** [comment]
@Memphiz @fetzerch could you check on iOS?

**Comment by Memphiz** [comment]
@MartijnKaijser how am i supposed to scroll down in the policy? (just trying it out on osx - its shown - but only the upper lines which fit into the sysinfo rect ...)

**Comment by ronie** [comment]
currently it autoscrolls, i'll look into adding a scrollbar.

**Comment by Memphiz** [comment]
works on ios too

**Comment by ronie** [comment]
cool, all platforms confirmed working then.

**Comment by MartijnKaijser** [comment]
Tested and looks fine to me. Jenkins build this please. 
@tamland was the implementation ok?

File: xbmc/windows/GUIWindowSystemInfo.cpp [hunk 5]
```diff
@@ -81,6 +82,23 @@ bool CGUIWindowSystemInfo::OnMessage(CGUIMessage& message)
...
+      else if (m_section == CONTROL_BT_HARDWARE)
+        SET_CONTROL_HIDDEN(13);
+      else if (m_section == CONTROL_BT_PVR)
+        SET_CONTROL_HIDDEN(13);
+      else if (m_section == CONTROL_BT_POLICY)
+      {
+        SET_CONTROL_LABEL(13, g_infoManager.GetLabel(SYSTEM_PRIVACY_POLICY));
+        SET_CONTROL_VISIBLE(13);
+      }
       return true;
```
Comment (phil65) [inline xbmc/windows/GUIWindowSystemInfo.cpp]: just
```
if (m_section > CONTROL_START && m_section < CONTROL_END)
{
 SET_CONTROL_HIDDEN(13);
}

```
should be fine.
For bonus points, use a #define for the textbox. :)

**Comment by tamland** [comment]
yeah, looks good

File: xbmc/windows/GUIWindowSystemInfo.cpp (hunk 5 above)
Comment (phil65) [inline xbmc/windows/GUIWindowSystemInfo.cpp]: oops, I meant
`if (m_section >= CONTROL_START && m_section <= CONTROL_BT_PVR)`

**Comment by ronie** [comment]
updated as @phil65 suggested.

Jenkins build this please.

File: xbmc/windows/GUIWindowSystemInfo.cpp [hunk 6]
```diff
@@ -182,6 +191,11 @@ void CGUIWindowSystemInfo::FrameMove()
     SetControlLabel(i++, "%s: %s", 19168, PVR_BACKEND_DELETED_RECORDINGS);  // Deleted and recoverable recordings
     SetControlLabel(i++, "%s: %s", 19025, PVR_BACKEND_TIMERS);
   }
+
+  else if (m_section == CONTROL_BT_POLICY)
+  {
+    SET_CONTROL_LABEL(40, g_localizeStrings.Get(12389));
```
Comment (phil65) [inline xbmc/windows/GUIWindowSystemInfo.cpp]: the 40 should be CONTROL_TB_POLICY  :)

Comment (phil65) [inline xbmc/windows/GUIWindowSystemInfo.cpp]: oops, sorry. false alarm.

**Comment by ronie** [comment]
> works on ios too

jenkins says no

**Comment by Memphiz** [comment]
i guess because Jenkins builds with cmake - i didn't ...

**Comment by Memphiz** [comment]
@ronie i think you Need to add privacy-policy.txt to this file here:

https://github.com/xbmc/xbmc/blob/master/project/cmake/installdata/ios/packaging.txt

and here

https://github.com/xbmc/xbmc/blob/master/project/cmake/installdata/osx/packaging.txt

because on osx it failed "silently" too: 

rsync: link_stat "/Users/Shared/jenkins/workspace/OSX-64/build/privacy-policy.txt" failed: No such file or directory (2)

@fetzerch any idea why osx Fails silently and ios correclty Fails the Job?

**Comment by ronie** [comment]
ok added, thx! 

Jenkins build this please.

**Pull request merged**